readyplayerme/rpm-unity-sdk-core
# [SDK-600] Feature/avatar template element

## [SDK-600](https://ready-player-me.atlassian.net/browse/SDK-600)

## Description

-   This PR includes a TemplateAvatar element that can be used to display avatar template icons and fire an event with the required data. It also includes a ButtonElement  class which should be usable in other cases. 
- it has I realize that we already have Asset and Avatar button classes in the sample but I wanted to keep this separate so as not to break the sample but also I think these simpler elements might be more re-usable. We can always update later as needed. 






## How to Test

-   easiest way to test is to checkout the **feature/template-avatar-element** branch in Unity-SDK repo and run the test scene in the TEMP folder. You should be able to:
1. run the scene
2. see the template buttons load
3. click a button
4. see selected icon
5. check console for print out on avatar ID and gender

![image](https://github.com/readyplayerme/rpm-unity-sdk-core/assets/7085672/a7702563-02c4-4b72-bb29-cee36d6e8ba0)



## Checklist








[SDK-600]: https://ready-player-me.atlassian.net/browse/SDK-600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
